src/App.js

```diff
@@ -25,7 +29,24 @@ class App extends Component {
                 { name: 'Jared', id: 0 },
                 { name: 'Adrian', id: 1 }
             ],
+            addingPlayer: false
         }
+        this.id = 1
+        this.addPlayer = this.addPlayer.bind(this);
+        this.addingPlayer = this.addingPlayer.bind(this);
+    }
+
+    addingPlayer(addingPlayer) {
+        this.setState({ addingPlayer: addingPlayer }) // making input field show up
+    }
+
+    addPlayer(name) {
+        this.id++;
+        this.setState({
+            originalPlayerList: this.state.originalPlayerList.concat({ name: name, id: this.id }),
+            playerList: this.state.playerList.concat({ name: name, id: this.id }),
```

Comment:
es6 has a more concise way of writing this that you might not be aware of called 'spread'  (and you can write a concat like this:
```js
[{ name, id: this.id }, ...this.state.playerList]
```
Notice I also just put `name`, if the variable name is the same as the key name it will implicitly assign it for you.
Another note on setState. I've found there can be sideeffects when calling this.state inside of a set state so it's typically best to assign the old state values to new variables outside of the set state
```js
let newPlayerList = this.state.playerList;
this.setState({ playerList: [{ name, id: this.id }, ...newPlayerList] });
```